scarlet/observation.py

```diff
@@ -146,11 +152,32 @@ def match(self, scene):
                 new_kernel_fft.append(np.fft.fft2(np.fft.ifftshift(kernel)))
             self.psfs_fft = np.array(new_kernel_fft)
 
-        # 4) divide obs.psf from scene.psf in Fourier space
+        return self
 
-        # [ 5) compute sparse representation of interpolation * convolution ]
+    @property
+    def B(self):
+        """Number of bands in the model
+        """
+        return self.shape[0]
+
+    @property
+    def Ny(self):
+        """Number of pixel in the y-direction
+        """
+        return self.shape[1]
+
+    @property
+    def Nx(self):
+        """Number of pixels in the x-direction
+        """
+        return self.shape[2]
 
     @property
+    def shape(self):
+        """Shape of the model.
+        """
+        return self._shape
+    @property
```

Comment:
I remain skeptical that we need to hide `images` at all. This is a left-over from pytorch, where they had different type, but I think we can remove that. @fred3m ?

Reply:
Same, it's an old thing I used for testing or possibly when I very briefly pytorched my code.

Reply:
This is a style issue. Because we initialize the observation with `images` and it is used to set the shape `(B, Ny, Nx)`, which also sets the image and PSF padding, having a user change `images` can have unexpected consequences that are difficult to diagnose. So I like to use a more C++ style approach and hide any variables that we really don't want the user to set, just to be safe.

Your argument is usually that the user should know better and they get what they deserve. I try to be a bit more forgiving *and* try to save us from emails from confused users with hard to diagnose issues. In my mind you only get what you deserve when you go around and monkey with hidden variables...